### DISCUSSION TITLE
add a layout for when kodi show pictures

### DISCUSSION THREAD

**Discussion by nico1080** [mattblovell/kodi_panel#34] [Ideas]
Currently, when kodi is displaying pictures, it will enable the status screen and it will display in the status fields "picture" so there is already somme detection of that.

Would it be possible to add a picture section with associated layout like we have for audio or video.

There is still the case when one play both music and picture. (but it will enter in the case of audio autoselection)

**Comment by mattblovell** [comment]
It is certainly possible.  I can extend the relevant "enable" variables and `update_display()`.

What would one show, though?  Looking at

https://kodi.wiki/view/InfoLabels

Section 8 only shows two players -- MusicPlayer and VideoPlayer.  I suppose there are a number of fields available under `ListItem.Picture*`:
- ListItem.PictureAperture
- ListItem.PictureAuthor
- ListItem.PictureCamMake
- ListItem.PictureCamModel
- ListItem.PictureDate

and so on.  Those could be interesting.

Presumably, if one is viewing pictures, there is a _much_ large display somewhere nearby. :)  This can go on the list, though.

**Reply by mattblovell** [reply]
One aspect I'm uncertain about, however, is that all of the `ListItem.Picture*` InfoLabels state that they are "for the selected picture".  Perhaps these InfoLabels only reflect the state of a selection within the Kodi GUI?  

In contrast, the MusicPlayer and VideoPlayer labels all talk about the "current song" and the "current movie" or "current Video".  

I worry whether there actually are InfoLabels for the "current picture".

**Reply by nico1080** [reply]
For the current picture, you have all the slideshow info label,  they are retrieved from the EXIF data of the played picture

see section 13:
https://kodi.wiki/view/InfoLabels#Slideshow

**Reply by nico1080** [reply]
> One aspect I'm uncertain about, however, is that all of the `ListItem.Picture*` InfoLabels state that they are "for the selected picture". Perhaps these InfoLabels only reflect the state of a selection within the Kodi GUI?
> 
> In contrast, the MusicPlayer and VideoPlayer labels all talk about the "current song" and the "current movie" or "current Video".
> 
> I worry whether there actually are InfoLabels for the "current picture".

"current picture" are `Slideshow*` infolabels
`ListItem.*` infolabels only apply to item selected by navigating the kodi GUI (and I do not see particuliar use of it for kodi panel application)

**Comment by mattblovell** [comment]
> For the current picture, you have all the slideshow info label, they are retrieved from the EXIF data of the played picture

Created Issue #35 to extend the existing framework to permit for screens during slideshows.

**Comment by mattblovell** [comment]
I have the changes ready, but note that I think an in-progress slideshow is not mutually exclusive with audio playback.  In `update_display()`, I prioritized invoking `audio_screens()` higher than the newly-added `slideshow_screens()`.
